.github/workflows/worker-ami-build.yaml

```diff
@@ -0,0 +1,123 @@
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: NextChapterSoftware/ec2-action-builder@v1
+        with:
+          github_token: ${{ secrets.RUNNER_PAT}}
+          aws_access_key_id: ${{ secrets.RUNNER_AWS_ACCESS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.RUNNER_AWS_SECRET_ACCESS_KEY }}
+          aws_region: "us-east-2"
+          ec2_instance_type: t3a.medium
+          ec2_ami_id: ami-03e93254c1055ef6a
+          ec2_subnet_id: "subnet-0a602071414b8b2eb"
+          ec2_security_group_id: "sg-02810c38b0dcb2462"
+          ec2_instance_ttl: 60                # Optional (default is 60 minutes)
+          ec2_spot_instance_strategy: BestEffort    # Other options are: SpotOnly, BestEffort, MaxPerformance 
+  build_worker_ami:
+    needs:
+      - start-runner
+    runs-on: ${{ github.run_id }}          
+    steps:              
+      - name: Checkout packer build repo
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.BUILD_PAT }}
+          repository: 'yambottle/dj-gitops'
+          path: 'dj-gitops'
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.10'
+      - run: pip install datajoint
+      - name: Call fetcher
+        env:
+          DJ_HOST: ${{ secrets.DJ_HOST }}
+          DJ_USER: ${{ secrets.DJ_USER }}
+          DJ_PASS: ${{ secrets.DJ_PASS }}
+          REPO_NAME: ${{ github.repository }}
+          WORKER_TYPE: ${{ inputs.worker_type }}
+          WORKFLOW_VERSION: ${{ inputs.workflow_version }}
+          STAGING: ${{ inputs.staging }}
+        run: |
+          export REPO_NAME=$(echo ${REPO_NAME} | cut -d "/" -f 2)
+          export ORG_NAME=$(echo ${REPO_NAME} | cut -d "_" -f 1)
+          export WORKFLOW_NAME=$(echo ${REPO_NAME} | cut -d "_" -f 2)
+          cd ${GITHUB_WORKSPACE}/dj-gitops/infrastructures/packer/worker_ami/inputs/
+          python fetcher.py
+      - name: Build worker AMI
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.BUILD_AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.BUILD_AWS_SECRET_ACCESS_KEY }}
+          STAGING: ${{ inputs.staging }}
+          REPO_NAME: ${{ github.repository }}
+          DOCKER_REGISTRY_HOST: ${{ secrets.DOCKER_REGISTRY_HOST }}
+          DOCKER_REGISTRY_REPO: ${{ secrets.DOCKER_REGISTRY_REPO }}
+          DOCKER_REGISTRY_USERNAME: ${{ secrets.DOCKER_REGISTRY_USERNAME }}
+          DOCKER_REGISTRY_PASSWORD: ${{ secrets.DOCKER_REGISTRY_PASSWORD }}
+        run: |
+          # Set up deploy SSH key
+          export REPO_NAME=$(echo ${REPO_NAME} | cut -d "/" -f 2)
+          echo -n "${{ secrets.DEPLOY_SSH_KEY_BASE64 }}" | base64 -d > ${GITHUB_WORKSPACE}/dj-gitops/infrastructures/packer/worker_ami/keys/${REPO_NAME}-deploy.pem
+          cd ${GITHUB_WORKSPACE}/dj-gitops/infrastructures/packer/worker_ami
+          packer init .
+          packer build -var-file ./inputs/${REPO_NAME}/${STAGING}.pkrvars.hcl .
```

Comment:
This `.pkrvars.hcl` file is created by the fetcher, correct?

Reply:
Yes, correct, also a `.env` for docker build